File: week-02/day4/student-filter.py

students = [
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Zsombor', 'age': 12, 'candies': 5}
]

def more_than(candy):

    y = ''
    for s in candy:
        if s['candies'] > 4:
            y = s['name']
            print(y)

more_than(students)

# create a function that takes a list of students and prints:
# - Who has got more candies than 4 candies

def avarage_candy(num_candy):

    all_candies = 0
    for i in num_candy:
        all_candies += i['candies']
    avarage = all_candies / len(students) #len is a must in order to get to know how many elements our list has
    return avarage

print(avarage_candy(students))

# create a function that takes a list of students and prints:
#  - how many candies they have on average

File: week-07/day3/student-filter.js
'use strict';

var students = [
        {'name': 'Rezso', 'age': 9.5, 'candies': 2},
        {'name': 'Gerzson', 'age': 10, 'candies': 1},
        {'name': 'Aurel', 'age': 7, 'candies': 3},
        {'name': 'Zsombor', 'age': 12, 'candies': 5}
]

// create a function that takes a list of students and logs:
// - Who has got more candies than 4 candies

// create a function that takes a list of students and logs: 
//  - how many candies they have on average

function moreCandy4(array){
    
    var moreCandy = new Array;
    
    array.forEach(function(element){
        if (element.candies > 4){
            moreCandy.push(element.name);
        };
    });
    return moreCandy;
};

console.log(moreCandy4(students));